lib/timetable.js

```diff
@@ -0,0 +1,70 @@
+'use strict'
+
+const {
+	parseDateTime,
+	parseStation,
+	normalizeEmpty,
+} = require('./util')
+const fetch = require('./fetch')
+
+const parseArrivalDeparture = (dep, station) => {
+	return {
+		tripId: dep.trip_uid, // @todo: `dep.ride_id`?
+		when: parseDateTime(dep.datetime),
+		cancelled: dep.is_cancelled === true,
+		direction: dep.direction,
+		lineName: dep.line_code,
+		stop: station,
+		route: dep.route.map(s => parseStation(s)),
+		routeAbbreviation: dep.through_the_stations,
+		hasTracker: normalizeEmpty(dep.has_tracker),
+	}
+}
+
+const _timetable = async (station, opt = {}) => {
+	opt = {
+		// @todo
+		when: null,
+		departureAfter: new Date(),
+		...opt,
+	}
+
+	if ('string' !== typeof station) {
+		if (station.type !== 'station') throw new Error('station is invalid')
+		station = station.id
+	}
+
+	const when = opt.when || opt.departureAfter
+	if (!when || !(when instanceof Date)) {
+		throw new Error('opt.departureAfter/opt.when must be null or a Date')
+	}
+
+	const res = await fetch(`network/station/${encodeURIComponent(station)}/timetable.json`, opt)
+	return res
+}
+
+const departures = async (station, opt = {}) => {
+	const t = await _timetable(station, opt)
+
+	if (!t.timetable || !Array.isArray(t.timetable.departures)) return []
+	station = parseStation(t.station)
+	return t.timetable.departures
+	.map(dep => parseArrivalDeparture(dep, station))
+}
+const arrivals = async (station, opt = {}) => {
+	const t = await _timetable(station, opt)
+
+	if (!t.timetable || !Array.isArray(t.timetable.arrivals)) return []
+	station = parseStation(t.station)
+	return t.timetable.arrivals
+	.map(dep => parseArrivalDeparture(dep, station))
+}
+
+departures.features = arrivals.features = { // require by fpti
+	...fetch.features,
+}
+
+module.exports = {
+	departures,
+	arrivals,
+}
```

Comment:
Please expose these methods under some different names, e.g. `departuresToday` or `arrivalsToday` since FPTI might use "normal" departures/arrivals methods the same way it currently has a `stopovers` method, which always takes some sort of date.